onnx/defs/nn/defs.cc

```diff
@@ -24,19 +24,31 @@ static std::string auto_pad_doc =
 
 namespace ONNX_NAMESPACE {
 
-void convPoolTypeAndShapeInference(InferenceContext& ctx, bool use_dilation, bool require_kernel_shape) {
+void convPoolTypeAndShapeInference(
+    InferenceContext& ctx,
+    bool use_dilation,
+    bool require_kernel_shape) {
   propagateElemTypeFromInputToOutput(ctx, 0, 0);
 
-  if (!hasNInputShapes(ctx, 2)) {
-    return;
+  if (require_kernel_shape) {
```

Comment:
I think this logic should be combined with lines 83-99 below: check if kernel_shape is available via attribute, and then determine whether we need 1 or 2 input shapes. (I think that the check on NumInputs below probably should be has-1-InputShape, right?)

Reply:
Sorry, it looks like you need the second input shape for the conv case even if kernel_shape attribute is specified. So, just choose 1 or 2 as required shapes based on require_kernel_shape